### PR TITLE
MM-41176 Don't send Rudder requests when blocked

### PR DESCRIPTION
**Pull request by hmhealey**
Currently, we repeatedly try to send telemetry to Rudder even if the ad blocker has blocked the initial setup. This leads to a ton of error spam in the dev tools when trying to debug things on community.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-41176
Closes https://github.com/mattermost/mattermost-server/issues/19380

#### Related Pull Requests
https://github.com/mattermost/focalboard/pull/2285

#### Release Note
```release-note
Stop attempting to send telemetry events when blocked by an ad blocker
```

### CODE REVIEW

File: components/root/root.jsx [hunk 1]
```diff
@@ -214,6 +214,10 @@ export default class Root extends React.PureComponent {
                 },
                 anonymousId: '00000000000000000000000000',
             });
+
+            rudderAnalytics.ready(() => {
+                Client4.setTelemetryHandler(new RudderTelemetryHandler());
```
Comment (crspeller) [inline components/root/root.jsx:219]: Do you also have to delete the invocation on 186?

**Review by crspeller — CHANGES_REQUESTED** [review]

File: components/root/root.jsx (hunk 1 above)
Comment (hmhealey) [reply components/root/root.jsx:219]: Good catch. That's what I get for doing a `git add -p` to skip some debugging code

**Review by crspeller — APPROVED** [review]
Yay!

**Comment by hmhealey** [comment]
@jespino Do you have a chance to look at this in the next day or two? If not, I can reassign

**Comment by jespino** [comment]
@hmhealey sure, I'll take a look today, sorry for the delay

**Pull request merged**